Assets/Scripts/TileMapDemoScripts/BaseTile.cs

```diff
@@ -106,33 +104,38 @@ public override void RefreshTile(Vector3Int position, ITilemap tilemap)
     {
         base.RefreshTile(position, tilemap);
     }
+}
 
-    public void setResources(int newResources)
+public class WaterTile : BaseTile
+{
+    public override void GetTileData(Vector3Int position, ITilemap tilemap, ref TileData tileData)
     {
-        resources = newResources;
+        base.GetTileData(position, tilemap, ref tileData);
+
+        // Reads from Assets/Resources folder
+        Sprite sprite = Resources.Load<Sprite>("water"); // Resources/water.png
```

Comment:
I believe you mentioned some hesitancy in using this "Resources.Load" method. After some surface level research, I found that it seems ok to be using this. Just while keeping in mind anything added to the "Resources" folder in our repo will be included in the build of our game. So just avoiding adding unnecessary stuff to that folder. (Looking at what's in there now, its all good).